File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/MedicinalProductIngredient.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class MedicinalProductIngredient : DomainResource, IResource
{
    public required Base.CodeableConcept Role { get; set; }
    public MedicinalProductIngredient_Substance? Substance { get; set; }
    public Base.Identifier? Identifier { get; set; }
    public Base.ResourceReference[]? Manufacturer { get; set; }
    public MedicinalProductIngredient_SpecifiedSubstance[]? SpecifiedSubstance { get; set; }
    public bool? AllergenicIndicator { get; set; }

    public class MedicinalProductIngredient_Substance : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public Base.ResourceReference[]? Strength { get; set; }
    }

    public class MedicinalProductIngredient_SpecifiedSubstance_Strength_ReferenceStrength : BackboneElement
    {
        public Base.CodeableConcept[]? Country { get; set; }
        public required Base.Ratio Strength { get; set; }
        public Base.CodeableConcept? Substance { get; set; }
        public string? MeasurementPoint { get; set; }
        public Base.Ratio? StrengthLowLimit { get; set; }
    }

    public class MedicinalProductIngredient_SpecifiedSubstance_Strength : BackboneElement
    {
        public Base.CodeableConcept[]? Country { get; set; }
        public required Base.Ratio Presentation { get; set; }
        public Base.Ratio? Concentration { get; set; }
        public string? MeasurementPoint { get; set; }
        public MedicinalProductIngredient_SpecifiedSubstance_Strength_ReferenceStrength[]? ReferenceStrength { get; set; }
        public Base.Ratio? PresentationLowLimit { get; set; }
        public Base.Ratio? ConcentrationLowLimit { get; set; }
    }

    public class MedicinalProductIngredient_SpecifiedSubstance : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public required Base.CodeableConcept Group { get; set; }
        public MedicinalProductIngredient_SpecifiedSubstance_Strength[]? Strength { get; set; }
        public Base.CodeableConcept? Confidentiality { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicinalProductIngredient.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class MedicinalProductIngredientSubstance(BackboneElement):
    code: CodeableConcept
    strength: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class MedicinalProductIngredientSpecifiedSubstanceStrengthReferenceStrength(BackboneElement):
    country: Optional[List[CodeableConcept]] = field(default_factory=list)
    strength: Ratio
    substance: Optional[CodeableConcept] = None
    measurement_point: Optional[str] = None
    strength_low_limit: Optional[Ratio] = None

@dataclass(kw_only=True)
class MedicinalProductIngredientSpecifiedSubstanceStrength(BackboneElement):
    country: Optional[List[CodeableConcept]] = field(default_factory=list)
    presentation: Ratio
    concentration: Optional[Ratio] = None
    measurement_point: Optional[str] = None
    reference_strength: Optional[List[MedicinalProductIngredientSpecifiedSubstanceStrengthReferenceStrength]] = field(default_factory=list)
    presentation_low_limit: Optional[Ratio] = None
    concentration_low_limit: Optional[Ratio] = None

@dataclass(kw_only=True)
class MedicinalProductIngredientSpecifiedSubstance(BackboneElement):
    code: CodeableConcept
    group: CodeableConcept
    strength: Optional[List[MedicinalProductIngredientSpecifiedSubstanceStrength]] = field(default_factory=list)
    confidentiality: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicinalProductIngredient(DomainResource):
    role: CodeableConcept
    substance: Optional[MedicinalProductIngredientSubstance] = None
    identifier: Optional[Identifier] = None
    manufacturer: Optional[List[Reference]] = field(default_factory=list)
    specified_substance: Optional[List[MedicinalProductIngredientSpecifiedSubstance]] = field(default_factory=list)
    allergenic_indicator: Optional[bool] = None